exporter/awsxrayexporter/translator/segment.go

```diff
@@ -338,14 +359,73 @@ func makeXRayAttributes(attributes map[string]string, indexedAttrs []string, ind
 				defaultMetadata[key] = value
 			}
 		}
-		if len(defaultMetadata) > 0 {
-			metadata["default"] = defaultMetadata
+
+		if storeResource {
+			resource.Attributes().ForEach(func(key string, value pdata.AttributeValue) {
+				key = "otel.resource." + key
+				annoVal := annotationValue(value)
+				_, indexed := indexedKeys[key]
+				if annoVal != nil && indexed {
```

Comment:
nit, it appears the ~repeated `storeResource` logic could occur for both `indexAllAttrs` values with an expanded condition.

Reply:
I think I can rework this indeed, thanks for noticing will update!